### ISSUE TITLE
Add tag filters to LabelMap

### ISSUE THREAD

**Issue by misaugstad** [ProjectSidewalk/SidewalkWebpage#3661]
##### Brief description of problem/feature
We can already filter by label type on LabelMap, but maybe it would be nice to filter by tag as well? :eyes: We already want to do a redesign of the filtering legend (#2849), partially because of how large and unruly it's become. We'd really have to think about how to redo the legend to work with so many different filtering options, but it would be really cool to include!

Here's a look at what LabelMap with it's legend looks like now.
![Screenshot from 2024-09-19 15-23-26](https://github.com/user-attachments/assets/bb1343c5-82f3-476a-a6cc-24e20d98aec6)

**Comment by misaugstad** [comment]
@jonfroehlich brought up that turning the legend into more of a sidebar is an option that we could consider. Generally we'd want to shy away from something like that, but it's worth consideration if we're covering a huge portion of the map with a legend anyway.

**Comment by jonfroehlich** [comment]
Right. This is quite common for complex UIs that offer filtering, etc.

And it's also standard UI for things like Google Maps.

![Image](https://github.com/user-attachments/assets/01a1df0a-fd3d-4a5f-bdbb-7bd0fe9d1877)

![Image](https://github.com/user-attachments/assets/1868994f-8b7a-4664-9cc0-af07907b46f9)

**Comment by Manjiridesigns** [comment]
We have this now. The select all/ deselect all changes based on the selections.
Eg :

<img width="1603" height="805" alt="Image" src="https://github.com/user-attachments/assets/369c5939-284e-48e9-a47f-8a7fe1f75e2b" />


View E2E prototype - [https://www.figma.com/design/hIZp7qtuaMr75tjvxfQxnt/LabelMap-Sidebar-Redesign?node-id=429-3&t=2SIuYg0yloRXhrkD-1]

**Comment by misaugstad** [comment]
@Manjiridesigns this ticket is about filtering by tags, which we haven't added and isn't included in your mocks.

There was a comment about turning the legend into a sidebar, which is what your mocks do, but it doesn't include filtering on tags.

**Comment by jonfroehlich** [comment]
I still need and want to filter by tags in Label Map. Maybe there is a "search" feature where we can just match on what the user types?

**Comment by Manjiridesigns** [comment]
Here is the final mock of how the filter with tag will look like. 

<img width="2405" height="1208" alt="Image" src="https://github.com/user-attachments/assets/fb94201a-d89e-4271-ba8e-b0ebaafb091b" />
<img width="2405" height="1208" alt="Image" src="https://github.com/user-attachments/assets/c601fea1-5e7a-44c5-a039-264fccb68ffc" />

Attaching the figma file for E2E prototype. 
[(https://www.figma.com/design/hIZp7qtuaMr75tjvxfQxnt/LabelMap-Sidebar-Redesign?node-id=429-3&t=lmiAW76zyf1GZp0k-1)]
